### DISCUSSION TITLE
Package security backports in 0.17

### DISCUSSION THREAD

**Discussion by haampie** [spack/spack#26869] [Ideas]
Taken from Slack, so that it remains here.

Can I propose backports for security fixes for some packages in the 0.17 release?

After `releases/v0.17` is branched off and `0.17.0` is released, we can branch off `backports/v0.17.1` from the release branch, and keep that open as a PR targeting `releases/v0.17` and cherry-pick security updates to say `openssl`, `curl`, ... and have gitlab pipelines run on that backports PR.

I can understand why you would want to freeze package versions in a 0.x Spack release, but security fixes are arguably an exception to that. `0.16` was released about a year ago, in the meantime `openssl` `1.1.1{i,j,k,l}` were released, all deprecating previous versions over security vulnerabilities. 

Also the e4s pipeline is about to drop the version preferences, we use `clingo` now, we have figured out `deprecated` and its concretization behavior in the last year... so actually testing security backports for packages is much easier in `0.17` than it used to be.

Another upside is that if we have that branch / PR right after the `0.17.0` release, we can immediately cherry-pick backport-worthy PRs to Spack core as they are merged, instead of that being an afterthought a few days ahead of the `0.17.1` release.

Even if we expect very disruptive / breaking changes to package.py's thanks to the compilers-as-package feature, I'm sure we can still backport, considering most changes to openssl etc are just a one-liner adding a new version.

---

[As a side note: w.r.t. the versions discussion, if we backport package.py's, we better minimize unnecessary breakage, so that we don't have to worry "should this range be interpreted in 0.17.x syntax, or in 0.18-dev syntax?")]

**Comment by cdfh** [comment]
I've always felt that spack would benefit from a 'stable' and 'development' branch, where all bugfixes get made in 'stable' and are then (immediately) merged forward into 'development'. As a user, I'm not eager to live on the cutting edge of spack development and instead favour stability over new features. In practice, this has meant that I have quite an old release checked out and that I rarely upgrade my entire repository, but rather, I manually checkout specific files/commits if and when I need them. This provides me with great stability, but on those occasions when I do want to bump forward to the latest and greatest spack release, there tends to be a lot of work to do. Also, it makes it harder for me to contribute back upstream: because my working branch is so old, I can't create PRs without first testing them on develop, which is no easy matter. If spack had a 'stable' branch which received only bugfixes, then it would make it much easier for me (and perhaps others?) to contribute upstream.

**Reply by haampie** [reply]
That is effectively what the `release/...` branches are for, but they don't get package backports
